docker/nuxeo-explorer-export-docker/Dockerfile

```diff
@@ -0,0 +1,43 @@
+# vim:set ft=dockerfile:
+ARG BASE_IMAGE
+
+FROM ${BASE_IMAGE}
+
+ARG VERSION
+ARG BUILD_TAG
+
+ARG USE_LOCAL_EXPLORER
+
+ARG CONNECT_EXPLORER_CLID
+ARG CONNECT_EXPLORER_URL
+ARG NUXEO_EXPLORER_PACKAGE
+
+ARG CONNECT_EXPORT_CLID
+ARG CONNECT_EXPORT_URL
+ARG EXPORT_PACKAGE_LIST
+
+LABEL org.nuxeo.explorer.export.version=${VERSION}
+LABEL org.nuxeo.explorer.export.build-tag=${BUILD_TAG}
+
+USER root
+
+# install explorer package
+COPY --chown=900:0 README.md target/nuxeo-platform-explorer-package-*.zip /packages/
+
+RUN if [ "${USE_LOCAL_EXPLORER}" = "true" ]; then \
+    /install-local-packages.sh /packages >&1; \
+  elif [ ! -z "${CONNECT_EXPLORER_CLID}" ]; then \
+    /install-packages.sh --clid "${CONNECT_EXPLORER_CLID}" --connect-url "${CONNECT_EXPLORER_URL}" ${NUXEO_EXPLORER_PACKAGE} >&1; \
+  fi
+
+# install reference packages
+RUN if [ ! -z "${CONNECT_EXPORT_CLID}" ] && [ ! -z "${EXPORT_PACKAGE_LIST// }" ]; then \
+    # replace "+" by space on package list because a conversion occurs when built with maven via a property
+    /install-packages.sh --clid "${CONNECT_EXPORT_CLID}" --connect-url "${CONNECT_EXPORT_URL}" ${EXPORT_PACKAGE_LIST//+/ } >&1; \
+  fi
```

Comment:
It might be clearer to have several "immutable" Dockerfiles rather than a generic one.
Using conditional statements, it is hard to have a reproducible image and the cache mechanism probably won't work since the step depends on build args.
The logic could be in the Groovy script since it seems to mostly depend on pipeline parameters.
WDYT?

Reply:
I used these logics so that the export image can be built even if there is no possible connection to connect (during maven build of the image, typically).
This image is never deployed anywhere else than the JX repo, so i don't see the point of generating "immutable" ones.
This is more like a "build, export and destroy" strategy, and we do not want any cache involved for it.

Reply:
OK, the point was more about readability and ability to debug an image that is always built following the same steps (reproducible). Just sounds weird to have the logic inside the Dockerfile rather than in the pipeline which drives the process.
Of course, if it's working and the image is volatile, it's fine.

Reply:
I indeed hesitated to have the logics inside the pipeline only, but having these logics in the Dockerfile still allows me to check the docker image build from maven in the explorer pakage use case, and to benefit from the "target connect URL switch" logics includes in install-packages script.
I confirm it is working :)